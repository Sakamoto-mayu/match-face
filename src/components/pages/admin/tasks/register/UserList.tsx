'use client'
import { userTable } from "@/const/userTable"
import CheckBox from "@/components/ui/checkbox/CheckBox"

const UserList = () => {
  const tableDefaultClassName = "border-2 border-deep-gray px-4 py-2 text-center"
  return (
    <div className="my-10">
      <table className="border-collapse border-2 border-deep-gray text-sm mx-auto">
        <thead className="bg-light-gray">
          <tr>
            <th className={`${tableDefaultClassName} w-12`}></th>
            <th className={`${tableDefaultClassName}`}>入社日</th>
            <th className={`${tableDefaultClassName}`}>職種</th>
            <th className={`${tableDefaultClassName}`}>状態</th>
            <th className={`${tableDefaultClassName} w-40`}>氏名</th>
          </tr>
        </thead>
        <tbody>
          {userTable.map((user) => (
            <tr key={`userId_${user.id}`}>
              <td className={`${tableDefaultClassName}`}>
                <CheckBox />
              </td>
              <td className={`${tableDefaultClassName}`}>{user.entry_date}</td>
              <td className={`${tableDefaultClassName}`}>{user.department}</td>
              <td className={`${tableDefaultClassName}`}>{user.user_status}</td>
              <td className={`${tableDefaultClassName}`}>{user.user_name}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  )
}

export default UserList
