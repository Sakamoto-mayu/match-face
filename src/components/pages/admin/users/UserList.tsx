"use client";
import { useState } from "react";
import { entry_data } from "@/const/userList";

const UserList = () => {
  const [sortedData, setSortedData] = useState(entry_data);

  // データを昇順もしくは降順に並べ替える
  const sortOptionChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const sortOption = event.target.value;
    if (sortOption === "入社日昇順") {
      const sortedData = [...entry_data].sort(function (a, b) {
        return a.entry_date < b.entry_date ? -1 : 1;
      });
      // console.log(sortedData);
      setSortedData(sortedData);
    } else {
      const sortedData = [...entry_data].sort(function (a, b) {
        return b.entry_date < a.entry_date ? -1 : 1;
      });
      // console.log(sortedData);
      setSortedData(sortedData);
    }
    return setSortedData;
  };

  return (
    <div className="flex justify-center mb-10">
      <div className="flex flex-col justify-center w-max">
        <div className="flex justify-end">
          <div>
            <label htmlFor="sortData">ソート：</label>
            <select
              name="userList"
              id="userList"
              className=" border border-deep-gary"
              onChange={sortOptionChange}
              defaultValue="入社日昇順"
              value="入社日昇順"
            >
              <option value="入社日昇順">入社日昇順</option>
              <option value="入社日降順">入社日降順</option>
            </select>
          </div>
        </div>
        <table className=" w-[720px] my-1">
          <thead>
            <tr className="border bg-deep-gray">
              <th className=" border py-3">入社日</th>
              <th className=" border">所属</th>
              <th className=" border">状態</th>
              <th className=" border">氏名</th>
            </tr>
          </thead>
          {sortedData.map((data: any, index: number) => (
            <tbody key={index}>
              <tr className="border text-center">
                <td className=" border py-3">{data.entry_date}</td>
                <td className=" border">{data.department}</td>
                <td className=" border">{data.user_status}</td>
                <td className=" border">{data.user_name}</td>
              </tr>
            </tbody>
          ))}
        </table>
      </div>
    </div>
  );
};

export default UserList;
