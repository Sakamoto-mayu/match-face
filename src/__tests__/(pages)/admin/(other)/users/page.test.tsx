import { render, screen, fireEvent } from "@testing-library/react";
import UsersPage from "@/app/(pages)/admin/(other)/users/page";
import React from "react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import UserList from "@/components/pages/admin/tasks/register/UserList";

describe("ユーザー管理画面のテスト", () => {
  beforeEach(() => {
    render(<UsersPage />);
  });
  it("スナップショット", async () => {
    const { container } = render(<UsersPage />);
    expect(container).toMatchSnapshot();
  });

  describe("UserListのテスト", () => {
    it("初期値に入社日昇順が設定されていることを確認", async () => {
      expect(screen.getByRole("option", { name: "入社日昇順" }).selected).toBe(
        true
      );
    });
    it("optionを入社日降順に変更する", () => {
      userEvent.click(screen.getByRole("option", { name: "入社日降順" }));
      expect(screen.getByRole("option", { name: "入社日降順" })).toBeInTheDocument()
    });
  });
});
