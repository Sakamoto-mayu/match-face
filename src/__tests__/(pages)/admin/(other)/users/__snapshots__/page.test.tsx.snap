// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ユーザー管理画面のテスト スナップショット 1`] = `
<div>
  <div
    class="flex justify-center"
  >
    <div
      class="border border-deep-gray w-max my-10"
    >
      <div
        class="flex items-center my-5"
      >
        <div>
          <input
            class="border w-[516px] mx-5 border-deep-gray"
            id="1"
          />
        </div>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-20 py-0.5"
        >
          検索
        </button>
      </div>
      <div
        class=" w-4/5 mx-5"
      >
        <select
          class="border"
          id="year"
          name="year"
        >
          <option
            value="2010"
          >
            2010
          </option>
          <option
            value="2011"
          >
            2011
          </option>
          <option
            value="2012"
          >
            2012
          </option>
          <option
            value="2013"
          >
            2013
          </option>
          <option
            value="2014"
          >
            2014
          </option>
          <option
            value="2015"
          >
            2015
          </option>
          <option
            value="2016"
          >
            2016
          </option>
          <option
            value="2017"
          >
            2017
          </option>
          <option
            value="2018"
          >
            2018
          </option>
          <option
            value="2019"
          >
            2019
          </option>
          <option
            value="2020"
          >
            2020
          </option>
          <option
            value="2021"
          >
            2021
          </option>
          <option
            value="2022"
          >
            2022
          </option>
          <option
            value="2023"
          >
            2023
          </option>
        </select>
        <label
          for="year"
        >
           年 
        </label>
        <select
          class="border"
          id="month"
          name="month"
        >
          <option
            value="1"
          >
            01
          </option>
          <option
            value="2"
          >
            02
          </option>
          <option
            value="3"
          >
            03
          </option>
          <option
            value="4"
          >
            04
          </option>
          <option
            value="5"
          >
            05
          </option>
          <option
            value="6"
          >
            06
          </option>
          <option
            value="7"
          >
            07
          </option>
          <option
            value="8"
          >
            08
          </option>
          <option
            value="9"
          >
            09
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="11"
          >
            11
          </option>
          <option
            value="12"
          >
            12
          </option>
        </select>
        <label
          for="month"
        >
           月
        </label>
      </div>
      <div
        class="my-5"
      >
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-20 mx-5 py-0.5"
        >
          Java
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-20 mx-5 py-0.5"
        >
          PHP
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-20 mx-5 py-0.5"
        >
          FR
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-20 mx-5 py-0.5"
        >
          CL
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-20 mx-5 py-0.5"
        >
          ML
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-20 mx-5 py-0.5"
        >
          QA
        </button>
      </div>
      <div
        class="mx-5 my-5"
      >
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-32 mr-5 py-0.5"
        >
          研修中
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-32 mr-5 py-0.5"
        >
          待機中
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm px-4 rounded-full border shadow-md w-32 mr-5 py-0.5"
        >
          アサイン中
        </button>
      </div>
    </div>
  </div>
  <div
    class="flex justify-center mb-10"
  >
    <div
      class="flex flex-col justify-center w-max"
    >
      <div
        class="flex justify-end"
      >
        <div>
          <label
            for="sortData"
          >
            ソート：
          </label>
          <select
            class=" border border-deep-gary"
            id="userList"
            name="userList"
          >
            <option
              selected=""
              value="入社日昇順"
            >
              入社日昇順
            </option>
            <option
              value="入社日降順"
            >
              入社日降順
            </option>
          </select>
        </div>
      </div>
      <table
        class=" w-[720px] my-1"
      >
        <thead>
          <tr
            class="border bg-deep-gray"
          >
            <th
              class=" border py-3"
            >
              入社日
            </th>
            <th
              class=" border"
            >
              所属
            </th>
            <th
              class=" border"
            >
              状態
            </th>
            <th
              class=" border"
            >
              氏名
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-23
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              田中太郎
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-24
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              山田花子
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-25
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              高橋和夫
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-26
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              佐藤ゆき
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-27
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              鈴木康子
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-28
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              山本亮
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-29
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              相田三雄
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="border text-center"
          >
            <td
              class=" border py-3"
            >
              2023-06-30
            </td>
            <td
              class=" border"
            >
              Java
            </td>
            <td
              class=" border"
            >
              待機中
            </td>
            <td
              class=" border"
            >
              中村隼人
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
